src/web/index.js

```diff
@@ -52,35 +56,13 @@ const postMessage = (type, data = {}) =>
     "*"
   );
 
-const updateExtensionThemeMiddleware = ({ getState }) => next => action => {
-  const returnValue = next(action);
-  const meta = action.meta || {};
-  if (!meta.skipAddon && themeChangeActions.includes(action.type)) {
-    postMessage("setTheme", { theme: selectors.theme(getState()) });
-  }
-  return returnValue;
-};
-
-const updateHistoryMiddleware = ({ getState }) => next => action => {
-  const returnValue = next(action);
-  const meta = action.meta || {};
-  if (!meta.skipHistory && themeChangeActions.includes(action.type)) {
-    const theme = selectors.theme(getState());
-    urlEncodeTheme(theme).then(url =>
-      window.history.pushState({ theme }, "", url)
-    );
-  }
-  return returnValue;
-};
-
 const composeEnhancers = composeWithDevTools({});
 
 const store = createAppStore(
   {},
   composeEnhancers(
     applyMiddleware(
-      updateExtensionThemeMiddleware,
-      updateHistoryMiddleware,
+      ...setupMiddleware({ postMessage, urlEncodeTheme, storage }),
```

Comment:
Bit of refactoring here, moved all the middleware into a separate module and added the stuff for image management between web app and add-on